File: test/cpp/test_read_events.cpp
// Your First C++ Program

#include <iostream>
#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_read_events" << std::endl;

    std::string filename = "";
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }

    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    std::vector<CPSGFile::EVGROUP> eventGroups = reader.getEventGroups();
    std::cout << "Event groups count:" << eventGroups.size() << "\n";
    for (int i = 0; i < eventGroups.size(); i++) {
        std::cout << "  " << eventGroups[i].Name << " index:" << i<< std::endl;
    }

    std::vector<CPSGFile::EVITEM> events = reader.getEvents();
    
    std::cout << "Events count:" << events.size() << "\n";
    int count = 0;
    for (const auto& event: events) {
        if (event.Name.compare("test_evt") == 0) {
            count++;
            //std::cout << "  " << event.Name << " (" << event.GroupName << ")  " << event.StartTime << "  " << event.TimeLength << std::endl;
        }
    }
    std::cout << count << std::endl;

    std::cout << "DONE" << std::endl;
    return 0;
}

File: test/python/test_read_events.py

from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader
    

reader = HarmonieReader.HarmonieReader()
print("Test: test_read_events")

filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

if (not res):
    print(f"ERROR Failed to open the file:{filename}")
    quit()
else:
    print("SUCCESS")

print("Reading events...")
events = reader.get_events()

if( len(events)==0 ):
    print("ERROR No events found")
    quit()
else:
    print(f"SUCCESS events count:{len(events)}")

print(f"Events count:{len(events)}")
for e in events:
    if e.name == "test_event":
        print(f" {e.name} ({e.group}) {e.start_time} {e.duration}")
    
event_groups = reader.get_event_groups()
print(f"Event groups count:{len(event_groups)}")
for g in event_groups:
    print(f" {g.name}")
    